platooning_control_ihp/src/platoon_control_ihp_worker.cpp

```diff
@@ -22,30 +22,98 @@
 namespace platoon_control_ihp
 {
    
-	PlatoonControlIHPWorker::PlatoonControlIHPWorker()
+	PlatoonControlWorkerIHP::PlatoonControlWorkerIHP()
     {
         pid_ctrl_ = PIDController();
         pp_ = PurePursuit();
 
     }
 
-    void PlatoonControlIHPWorker::updateConfigParams(PlatoonControlIHPPluginConfig new_config)
+    void PlatoonControlWorkerIHP::updateConfigParams(PlatooningControlIHPPluginConfig new_config)
     {
         ctrl_config_ = new_config;
         pid_ctrl_.config_ = new_config;
         pp_.config_ = new_config;
     }
 
-	double PlatoonControlIHPWorker::getLastSpeedCommand() const {
+	double PlatoonControlWorkerIHP::getLastSpeedCommand() const {
         return speedCmd_;
     }
 
-    void PlatoonControlIHPWorker::setCurrentPose(const geometry_msgs::PoseStamped msg)
+    void PlatoonControlWorkerIHP::setCurrentPose(const geometry_msgs::PoseStamped msg)
 	{
 		current_pose_ = msg.pose;
 	}
 
-    void PlatoonControlIHPWorker::generateSpeed(const cav_msgs::TrajectoryPlanPoint& point)
+    // UCLA: Put the IHP gap regulation at low level control plug-in
+    double PlatoonControlWorkerIHP::getIHPDesPosFollower(double leaderCurrentPosition)
```

Comment:
I think I need a little more explanation of how this function calculates the desired position for the host vehicle.

Reply:
This is a gap regulation algorithm that uses the leader's info, preceding vehicle's info and host info to calculate a desired position.  This was implemented in the previous stages of the project. I will find the source code and make a documentation with Xin.